tirtha_bk/tirtha/views.py

```diff
@@ -231,25 +257,56 @@ def _signin(token, create=False):
 
 
 @require_GET
-def googleAuth(request):
-    """
-    Sets auth token cookie post Google Auth.
+def google_login(request):
+
+    # build a full authorize callback uri
 
+    new_state = str(uuid.uuid4())  # Generate a new UUID
+    redirect_uri = request.build_absolute_uri('/signin-google') #TODO: build the redirect uri as per configugered in the GOOGLE BACKEND
+    request.session['auth_state'] = new_state
+
+    return oauth.google.authorize_redirect(request, redirect_uri, state = new_state)
+
+@require_GET
+def logout(request):
+    #user can signout after signing in if the user wants to
+    # build a logout button
     """
-    token = request.GET["token"]
-    output, contrib = _signin(token, create=True)
+    Pops the user from the session post logout.
+    The user is redirected to the index page.
+    """
+    request.session.pop('auth_token', None)
+    return redirect('index')
+
 
-    context = {"output": output, "banned": False, "blur": False}
+@require_GET
+def tokenAuth(request):
+    """
+    Sets auth token cookie post token-based authentication.
+    """                                
+    # stored_state = request.session.get('auth_state')
+    # received_state = request.GET.get('state')
 
-    # Store JWT in session
-    if contrib is not None:
+    try:
+        # Get token from OAuth provider
+        token = oauth.google.authorize_access_token(request)
+        token = dict(token)
         request.session["auth_token"] = token
-        request.session.set_expiry(0)  # Session expires when browser closes
 
-        if contrib.banned or not contrib.active:
-            context.update({"blur": True})
+        # Call _signin function to handle token-based authentication
```

Comment:
L296 - L304 are unneeded now. As `output`, `contrib` and `context` are not being used anywhere. `index()` handles these cases anyway. These can be removed. Keep the Exception, but instead of setting `context`, just set `request.session["auth_token"]` to `None`.